Update proposal / voting documentation (#2877)
dfinity/portal: docs/developer-docs/daos/nns/nns-app-quickstart.mdx

The NNS uses a form of liquid democracy to address these challenges.
To aid in this, the NNS supports **neuron following **.
A neuron can be configured to vote automatically by following the voting decision made by a group of other neurons.
Following is done on the basis of topics and a neuron can follow up to 15 neurons on each topic.

### Following rules
A neuron can choose to follow the voting decision for different proposal topics like governance or subnet management. For example, if a neuron has been configured to follow a group's decision for governance proposals, if the group votes to adopt a governance proposal with more than half of the followees voting in favor of adoption, the neuron following that group will vote to approve the proposal as well. If the neuron has been configured to follow a group's decision for subnet management proposals and the group votes to reject a subnet management proposal with at least half of the followees voting to reject, the neuron following the group will vote to reject the proposal.

:::caution

> Your neuron will follow when there are enough votes to make a call (whether it is a ADOPT or REJECT). Your neuron will only cast a "Yes"-vote if **more than half** of the neurons within the group voted Yes. Your neuron will only caes a "No"-vote if **at least half** of the neurons within the group voted No. For example, if you follow 10 neurons on a topic:

Reviewer: ```suggestion
Your neuron will follow when there are enough votes to make a call (whether it is a ADOPT or REJECT). Your neuron will only cast a "Yes"-vote if **more than half** of the neurons within the group voted Yes. Your neuron will only cast a "No"-vote if **at least half** of the neurons within the group voted No. For example, if you follow 10 neurons on a topic:
```

a. Your neuron only votes to adopt if at least 6 out of the 10 neurons vote to adopt.

b. Your neuron only votes to reject a proposal if at least 5 out of the 10 reject the proposal.

c. If less than 5 neurons (out of your 10 followed) vote at all, your neuron will not cast any vote.

This should be considered when choosing to follow multiple neurons on a topic.

> It might be better to follow fewer neuron and it can be beneficial to follow an odd number of neurons than even numbers.

Reviewer: ```suggestion
It might be better to follow fewer neurons, and it can be more beneficial to follow an odd number of neurons than an even number.
```

:::

### Following "catch-all"
In addition to setting following for individual topics, neurons may also define a catch-all follow rule which is triggered for proposals with topics for which no other explicit follow rule has been defined.

> However, the catch-all follow rule does not include SNS & neuron's fund or governance proposals.

Reviewer: ```suggestion
However, the catch-all follow rule does not include SNS & neurons' fund or governance proposals.
```

:::caution
The catch-all follow rule is only applied for proposal topics with no other explicit choice. For example, let's assume neuron A follows neuron B for the catch-all and neuron C on the topic subnet management. If B votes on a subnet management proposal but C does not, no vote will be cast for A.

> The catch-all follow rule does not apply for SNS & neuron's fund or governance proposals.

Reviewer: ```suggestion
The catch-all follow rule does not apply for SNS & neurons' fund or governance proposals.
```
